[geotrans] update to version 3.8 (on-hold waiting for CI/VM port 21 for ftp to be opened or alternate host site)

StarGate-One opened the pull request:
1. Added new files to CMakeLists.txt required by version 3.8
2. Remove old style CONTROL file used by vcpkg
3. Updated portfile.cmake for version 3.8
   - download location now a ftp server
   - requires CI/VM to be updated to open ftp port 21 or alternate host site
4. Add new style vcpkg.json file used by vcpkg

**Describe the pull request**

- What does your PR fix? Fixes #13241 Replaces PR #13243

- Which triplets are supported/not supported? Only windows/linux dynamic x86/x64 supported.
- Have you updated the CI baseline? No updates

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)? Yes best of my knowledge.

StarGate-One committed:
Update port geotrans to version 3.8
1. Added new files to CMakeLists.txt required by version 3.8
2. Remove old style CONTROL file used by vcpkg
3. Updated portfile.cmake for version 3.8
4. Add new style vcpkg.json file used by vcpkg

StarGate-One commented:
@BillyONeal 
Please could you stop the CI/VM runs for this PR as they all will fail until the CI/VM ftp port 80 is opened.
Thank you...

BillyONeal commented:
Quoting the other PR:

>Although the ftp server is very slow, it takes about an hour to download the 156mb file

We might need to disable this port in our CI if it's going to hang the agents for an hour. Maybe when we can cache inputs...

StarGate-One commented:
> Quoting the other PR:
> 
> > Although the ftp server is very slow, it takes about an hour to download the 156mb file
> 
> We might need to disable this port in our CI if it's going to hang the agents for an hour. Maybe when we can cache inputs...

I have no problem with it -- I have searched to see if I could find if someone else already had it cached, but could not find it anywhere.

I did previously find where the CONAN project mentions the same issue https://github.com/conan-io/conan-center-index/issues/1779 .

It does look like one can download/host/redistribute the software based on the downloaded MSP Geotrans Terms of Use from their site: [MSP_Geotrans_Terms_Of_Use.txt](https://github.com/microsoft/vcpkg/files/5498175/MSP_Geotrans_Terms_Of_Use.txt)

BillyONeal commented:
Despite the port being now open looks like upstream is too unreliable. If it is acceptable to you can you add =skip entries to `ci.baseline.txt` with a comment about the slow/unreliable upstream?

StarGate-One committed:
Add skip to CI baseline. Add slow warning messages to portfile

StarGate-One committed:
Refine warning message about long download times

StarGate-One commented:
@BillyONeal @JackBoosY 
- I updated the CI to skip x64-linux, x64-windows and x86-windows as those are the only 3 we are building the port for. The rest of the standard triplets are handled in the file vcpkg.json supports section.
- I also updated the warning message(s) in the file portfile.cmake stating long download time for source.

JackBoosY commented:
@StarGate-One Can we use geotrans other mirrors?

StarGate-One commented:
> @StarGate-One Can we use geotrans other mirrors?

@JackBoosY 
- I have not found a mirror with version 3.8, although there are a few github repos with version 3.7
- It looks, based on this document [MSP_Geotrans_Terms_Of_Use.txt](https://github.com/microsoft/vcpkg/files/5740940/MSP_Geotrans_Terms_Of_Use.txt), it can be hosted in other repos as long as the license guidelines are followed and it is made clear the government and/or the NGA does not endorse or recommend any software it is used in...

JackBoosY commented on ports/geotrans/vcpkg.json:
```diff
@@ -0,0 +1,7 @@
+  "version-string": "3.8",
+  "description": "GEOTRANS is an application that allows you to convert geographic coordinates among a wide variety of coordinate systems, map projections, grids, and datums. GEOTRANS runs in Microsoft Windows and LINUX environments.",
+  "homepage": "https://earth-info.nga.mil/GandG/update/index.php?action=home",
+  "supports": "(((x86 | x64) & windows & !static) | ((x86 | x64) & linux)) & !uwp & !arm & !arm64 & !osx & !android"
```
```suggestion
```

JackBoosY replied:
Since you've added the items on the baseline, we don't need this.

StarGate-One replied:
@JackBoosY /cc @BillyONeal @ras0219-msft @vicroms @strega-nil 

**Question: What is the direction going forward - should we be using the `supports` tag in the `CONTROL or vcpkg.json` files or be using the `ci.baseline.txt` file?**
- My understanding is as follows:
   - The `ci.baseline.txt` file is only for the `CI pipeline`.
   -  The `CONTROL or vcpkg.json` works in both the `CI pipeline` and the `user experience` on the user's system.
   - We only added the 3 supported triplets with the `skip` directive for the benefit of the `CI pipeline`, because we do not want the `CI pipeline` to be held up with slow downloads that will eventually fail anyway due to ftp timeout.
   - If we remove the `supports` tag as suggested then the ports not supported will fail in both the `CI pipeline` and the end users system.
- Previously I had all the unsupported triplets in the `ci.baseline.txt` file and only the supported triplets listed in the `supports` tag of the `vcpkg.json` file, 
- My next instructions were to remove all from `ci.baseline.txt` and add `supported\unsupported` to the `supports` tag of the `vcpkg.json` file.
- My next instructions was to add the `supported triplets` back to the `ci.baseline.txt` file with the `skip` directive to allow the `CI pipeline` to bypass these triplets due to the excessive time needed to ftp the file from the origin server.

Maybe a discussion on what is the preferred direction going forward?

BillyONeal replied:
>The ci.baseline.txt file is only for the CI pipeline.

Correct.

>The CONTROL or vcpkg.json works in both the CI pipeline and the user experience on the user's system.

It *should* work on the end user's system but doesn't yet :/

>We only added the 3 supported triplets with the skip directive for the benefit of the CI pipeline, because we do not want the CI pipeline to be held up with slow downloads that will eventually fail anyway due to ftp timeout.

Yes.

I think the PR is correct.
